---
pinger_negate:
  param: /mission/acoustic_pinger/negate
  options: [true, false]
  description: pinger negate
  default: false

explorer_next:
  param: /mission/explorer/next
  options: [ALL, BuoyField, CoralSurvey, FindBreak,
            AcousticPinger, Shooter, Scan_The_Code,
            Gate_1, Gate_2, Gate_3, Dock]
  # yamllint disable-line rule:line-length
  description: The next unknown object to search for when no missions are running, defaults to furthest away (ALL)
  default: ALL

totem_direction_1:
  param: /mission/totems/direction_1
  options: [CLOCKWISE, COUNTER-CLOCKWISE, false]
  description: >
    What direction the Find Totems and Avoid Obstacles
    mission should circle first
    if False, no more totems should be circled
totem_color_1:
  param: /mission/totems/color_1
  options: [RED, GREEN, YELLOW, BLUE]
  # yamllint disable-line rule:line-length
  description: What color totem the Find Totems and Avoid Obstacles mission should circle first

totem_direction_2:
  param: /mission/totems/direction_2
  options: [CLOCKWISE, COUNTER-CLOCKWISE, false]
  description: >
    What direction the Find Totems and Avoid Obstacles
    mission should circle second
    if False, no more totems should be circled
totem_color_2:
  param: /mission/totems/color_2
  options: [RED, GREEN, YELLOW, BLUE]
  # yamllint disable-line rule:line-length
  description: What color totem the Find Totems and Avoid Obstacles mission should circle second

totem_direction_3:
  param: /mission/totems/direction_3
  options: [CLOCKWISE, COUNTER-CLOCKWISE, false]
  description: >
    # yamllint disable-line rule:line-length
    What direction the Find Totems and Avoid Obstacles
    mission should circle third
    if False, no more totems should be circled
totem_color_3:
  param: /mission/totems/color_3
  options: [RED, GREEN, YELLOW, BLUE]
  # yamllint disable-line rule:line-length
  description: What color totem the Find Totems and Avoid Obstacles mission should circle third

totem_direction_4:
  param: /mission/totems/direction_4
  options: [CLOCKWISE, COUNTER-CLOCKWISE, false]
  description: >
    What direction the Find Totems and Avoid Obstacles
    mission should circle fourth
    if False, no more totems should be circled
totem_color_4:
  param: /mission/totems/color_4
  options: [RED, GREEN, YELLOW, BLUE]
  # yamllint disable-line rule:line-length
  description: >
    What color totem the Find Totems and Avoid Obstacles
    mission should circle fourth

detect_deliver_target:
  param: /mission/detect_deliver/target
  options: [BIG, SMALL]
  description: Which target to shoot at in detect deliver
  default: BIG

detect_deliver_color:
  param: /mission/detect_deliver/Color
  options: [RED, GREEN, BLUE, ANY]
  description: The color of the symbol to shoot at in detect deliver
detect_deliver_shape:
  param: /mission/detect_deliver/Shape
  options: [CIRCLE, TRIANGLE, CROSS, ANY]
  description: The shape of the symbol to shoot at in detect deliver

scan_the_code_color1:
  param: /mission/scan_the_code/color1
  options: [RED, GREEN, YELLOW, BLUE]
  description: The first color in the 3 color sequence of scan the code
scan_the_code_color2:
  param: /mission/scan_the_code/color2
  options: [RED, GREEN, YELLOW, BLUE]
  description: The second color in the 3 color sequence of scan the code
scan_the_code_color3:
  param: /mission/scan_the_code/color3
  options: [RED, GREEN, YELLOW, BLUE]
  description: The third color in the 3 color sequence of scan the code

coral_survey_shape1:
  param: /mission/coral_survey/shape1
  options: [CIRCLE, TRIANGLE, CROSS]
  # yamllint disable-line rule:line-length
  description: The shape identified in quadrant I of the Underwater Shape Identification challenge
coral_survey_shape2:
  param: /mission/coral_survey/shape2
  options: [CIRCLE, TRIANGLE, CROSS]
  # yamllint disable-line rule:line-length
  description: The shape identified in quadrant II of the Underwater Shape Identification challenge
coral_survey_shape3:
  param: /mission/coral_survey/shape3
  options: [CIRCLE, TRIANGLE, CROSS]
  # yamllint disable-line rule:line-length
  description: The shape identified in quadrant III of the Underwater Shape Identification challenge
coral_survey_shape4:
  param: /mission/coral_survey/shape4
  options: [CIRCLE, TRIANGLE, CROSS]
  # yamllint disable-line rule:line-length
  description: The shape identified in quadrant IV of the Underwater Shape Identification challenge

coral_survey_search_first:
  param: /mission/coral_survey/search_first
  options: [1, 2, 3, 4]
  # yamllint disable-line rule:line-length
  description: One of the quadrants to search for shapes in during the coral survey mission
coral_survey_search_second:
  param: /mission/coral_survey/search_second
  options: [1, 2, 3, 4]
  # yamllint disable-line rule:line-length
  description: The other quadrant to search for shapes in during the coral survey mission

find_the_break_markers:
  param: /mission/find_the_break/markers
  options: [1, 2, 3, 4, 5]
  # yamllint disable-line rule:line-length
  description: The number of orange wall segments between the yellow breaks in the find the break challenge

acoustic_pinger_active:
  param: /mission/acoustic_pinger/active
  options: [RED-WHITE, WHITE-WHITE, WHITE-GREEN]
  description: >
    The two buoys the active pinger is between in the
    acoustic pinger transit.
acoustic_pinger_active_index:
  param: /mission/acoustic_pinger/active_index
  options: [1, 2, 3]
  description: >
    Which gate (as identified in the database) the pinger was found in.
    For example, 2 means Gate_2 is the location of the pinger
acoustic_pinger_active_index_correct:
  param: /mission/acoustic_pinger/active_index_correct
  options: [1, 2, 3]
  description: >
    Which gate the pinger was found in. Adjusted for rules (RED=1)

dock_shape_1:
  param: /mission/identify_dock/shape_1
  options: [CIRCLE, TRIANGLE, CROSS, ANY]
  description: >
    The shape of the first bay to dock in during the identify
    Symbols and Dock mission. If set to ANY, mission should dock based on the
    color parameter and set this parameter during the mission
dock_color_1:
  param: /mission/identify_dock/color_1
  options: [RED, GREEN, BLUE, ANY]
  description: >
    The color of the first bay to dock in during the identify Symbols and Dock
    mission. If set to ANY, mission should dock based on the shape parameter
    and set this parameter during the mission

dock_shape_2:
  param: /mission/identify_dock/shape_2
  options: [CIRCLE, TRIANGLE, CROSS, ANY]
  description: >
    # yamllint disable-line rule:line-length
    The shape of the second bay to dock in during the
    identify Symbols and Dock mission. If set to ANY,
    mission should dock based on the color parameter and set this
    parameter during the mission
dock_color_2:
  param: /mission/identify_dock/color_2
  options: [RED, GREEN, BLUE, ANY]
  description: >
    # yamllint disable-line rule:line-length
    The color of the second bay to dock in during the
    identify Symbols and Dock mission.
    If set to ANY, mission should dock based on the shape
    parameter and set this parameter during the mission
